uvicorn/subprocess.py

```diff
@@ -47,12 +40,13 @@ def subprocess_started(config, target, sockets, stdin_fileno):
                be the `Server.run()` method.
     * sockets - A list of sockets to pass to the server. Sockets are bound once
                 by the parent process, and then passed to the child processes.
-    * stdin_fileno - The file number of sys.stdin, so that it can be reattached
-                     to the child process.
     """
     # Re-open stdin.
-    if stdin_fileno is not None:
-        sys.stdin = os.fdopen(stdin_fileno)
+    # This is required for some debugging environments.
+    try:
+        sys.stdin = os.fdopen(0)
```

Comment:
I think I'm a bit confused. Does this version fixes the issue you were seeing? Asking because I would expect stdin.fileno() to return 0, as I believe is traditional on Unix systems, so this code would do the same as initially... ?

Were you able to see the value of the file descriptor that was throwing "Bad file descriptor"?

Do we need to add `os.dup(stdin_fileno)` to create a copy rather than reusing it directly?

Reply:
>  Does this version fixes the issue you were seeing?
Yes, it does.

> I would expect stdin.fileno() to return 0, as I believe is traditional on Unix systems
I had exactly the same expectation.

> Were you able to see the value of the file descriptor that was throwing "Bad file descriptor"?
Yes, I was. In the original setup, `stdin.fileno()` returns values greater than 2 (such as 9, 10 or 11; I mentioned that in the [original report](https://github.com/encode/uvicorn/issues/877#issuecomment-742133901)). 

I am not sure if the issue runs deeper than that (for example, an issue with the way `stdin.fileno()` retrieves the file descriptor for `stdin`), but the fix definitely works - I've been using my branch for a couple of weeks now.

I would be curious to see if you can replicate the issue - there is an MRE in the [bug report](https://github.com/encode/uvicorn/issues/877#issuecomment-742133901).